book-content/chapters/06-objects.md

```diff
@@ -0,0 +1,1386 @@
+
+In this example, we have a base `Album` interface that will be extended into `StudioAlbum` and `LiveAlbum` interfaces that allow us to provide more specific details about an album:
+
+```typescript
+interface Album {
+  title: string;
+  artist: string;
+  releaseYear: number;
+}
+
+interface StudioAlbum extends Album {
+  studio: string;
+  producer: string;
+}
+
+interface LiveAlbum extends Album {
+  concertVenue: string;
+  concertDate: Date;
+}
+```
+
+This structure allows us to create more specific album representations with a clear inheritance relationship:
+
+```typescript
+const americanBeauty: StudioAlbum = {
+  title: "American Beauty",
+  artist: "Grateful Dead",
+  releaseYear: 1970,
+  studio: "Wally Heider Studios",
+  producer: "Grateful Dead and Stephen Barncard",
+};
+
+const oneFromTheVault: LiveAlbum = {
+  title: "One from the Vault",
+  artist: "Grateful Dead",
+  releaseYear: 1991,
+  concertVenue: "Great American Music Hall",
+  concertDate: new Date("1975-08-13"),
+};
+```
+
+Just as adding additional `&` operators add to an intersection, it's also possible for an interface to extend multiple other interfaces by separating them with commas:
+
+```typescript
+interface BoxSet extends StudioAlbum, LiveAlbum {
+  numberOfDiscs: number;
+}
+```
+
+### Intersections vs `interface extends`
+
+We've now covered two separate TypeScript syntaxes for extending object types: `&` and `interface extends`. So, which is better?
+
+You should choose `interface extends` for two reasons.
+
+#### Better Errors When Merging Incompatible Types
+
+We saw earlier that when you intersect two object types with an incompatible property, TypeScript will resolve the property to `never`:
+
+```typescript
+type User1 = {
+  age: number;
+};
+
+type User2 = {
+  age: string;
+};
+
+type User = User1 & User2;
+```
+
+When using `interface extends`, TypeScript will raise an error when you try to extend an interface with an incompatible property:
+
+```ts twoslash
+// @errors: 2430
+interface User1 {
+  age: number;
+}
+
+interface User extends User1 {
+  age: string;
+}
+```
+
+This is very different because it actually sources an error. With intersections, TypeScript will only raise an error when you try to access the `age` property, not when you define it.
+
+So, `interface extends` is better for catching errors when building out your types.
+
+#### Better TypeScript Performance
+
+When you're working in TypeScript, the performance of your types should be at the back of your mind. In large projects, how you define your types can have a big impact on how fast your IDE feels, and how long it takes for `tsc` to check your code.
+
+`interface extends` is much better for TypeScript performance than intersections. With intersections, the intersection is recomputed every time it's used. This can be slow, especially when you're working with complex types.
```

Comment:
Intersections are heavily cached too. Some resolutions sometimes (maybe? I'm not sure) can be recomputed but in many cases, they won't.